### PR TITLE
Feat: adapt to peopledoc org and custom to fit SRE needs

### PR DESCRIPTION
**Pull request by StessyMougeot**
- Add codeowners
- Enhance role to fit needs
- Add molecules tests

PR mood:

### CODE REVIEW

File: meta/.galaxy_install_info [hunk 1]
```diff
@@ -0,0 +1,2 @@
+install_date: Tue Feb  2 09:24:41 2021
```
Comment (SwannCroiset) [inline meta/.galaxy_install_info]: this file is required ?

File: tasks/main.yml [hunk 2]
```diff
@@ -23,21 +23,88 @@
...
+    state: present
+    groups: adm
+    system: True
+    append: True
+
+- name: Ensure unit directory is present
+  file:
+    path: /etc/systemd/system/td-agent-bit.service.d
+    state: directory
+    mode: '0755'
```
Comment (RymRabehi) [inline tasks/main.yml]:
```suggestion
    mode: '0755'
```
```suggestion
    mode: '0644'
```

File: tasks/main.yml [hunk 3]
```diff
@@ -23,21 +23,88 @@
...
+    state: present
+    system: True
+
+- name: "Create user with adm group"
+  user:
+    name: td-agent-bit
+    create_home: False
+    group: td-agent-bit
+    state: present
+    groups: adm
```
Comment (SwannCroiset) [inline tasks/main.yml]: groups should be a list.
I'd add a default list into default/main.yml to be able to override it (we'll probably need to add this users to other groups ..)

File: tasks/main.yml [hunk 4]
```diff
@@ -23,21 +23,88 @@
...
+    system: True
+    append: True
+
+- name: Ensure unit directory is present
+  file:
+    path: /etc/systemd/system/td-agent-bit.service.d
+    state: directory
+    mode: '0755'
+
+- name: Add unit override.conf file
```
Comment (SwannCroiset) [inline tasks/main.yml:48]: don't you need to reload systemd ?

File: tasks/main.yml [hunk 5]
```diff
@@ -23,21 +23,88 @@
...
     path: "{{ fluentbit_dbs_path }}"
     state: directory
     mode: 0755
+    owner: td-agent-bit
+    group: td-agent-bit
   when: fluentbit_dbs_path | length > 0
 
+- name: "Create logs directory."
+  file:
+    path: "{{ fluentbit_log_path }}"
```
Comment (SwannCroiset) [inline tasks/main.yml:69]: what is the default of fluentbit_log_path ?

File: tasks/main.yml [hunk 6]
```diff
@@ -23,21 +23,88 @@
...
     dest: /etc/td-agent-bit/td-agent-bit.conf
     mode: 0644
   notify: "Restart Fluentbit."
+
+- name: Allow dd-agent to monitor fluentbit on port 2020
+  ansible.builtin.iptables:
+    chain: OUTPUT
+    protocol: tcp
+    destination_port: '2020'
+    uid_owner: dd-agent
```
Comment (SwannCroiset) [inline tasks/main.yml]: what's happen if dd-agent doesn't exist ?
IMHO should be conditional

File: files/functions.lua [hunk 7]
```diff
@@ -0,0 +1,9 @@
+function cb_ms_to_s(tag, timestamp, record)
```
Comment (RymRabehi) [inline files/functions.lua]: What does cb stand for?

File: tasks/main.yml [hunk 8]
```diff
@@ -23,21 +23,88 @@
...
+
 - name: "Create main Fluentbit configuration."
   template:
     src: td-agent-bit.conf.j2
     dest: /etc/td-agent-bit/td-agent-bit.conf
     mode: 0644
   notify: "Restart Fluentbit."
+
+- name: Allow dd-agent to monitor fluentbit on port 2020
+  ansible.builtin.iptables:
```
Comment (SwannCroiset) [inline tasks/main.yml]: ansible.builtin.iptables: this module has pre-requisits and I'm not sure we have iptables everywhere (on prod/rax-priv envs we have only security groups no iptables). it deserves to test this situation

File: tasks/main.yml [hunk 9]
```diff
@@ -23,21 +23,88 @@
...
+    chain: OUTPUT
+    protocol: tcp
+    destination_port: '2020'
+    uid_owner: dd-agent
+    jump: ACCEPT
+    comment: Accept dd-agent to monitor td-agent-bit:2020
+
+- name: Unallow all except dd-agent to monitor fluentbit on port 2020
```
Comment (SwannCroiset) [inline tasks/main.yml]: this negative rule should be enough ? :thinking:

**Comment by RymRabehi** [comment]
https://github.com/peopledoc/ansible-role-fluentbit#requirements
> This role is only tested on Debian 10.x (Buster).

Did you tested it on Debian 9?

Also, shouldn't we add molecule tests? What do you think?

File: files/functions.lua (hunk 7 above)
Comment (StessyMougeot) [reply files/functions.lua]: I don't remember… bad naming so

File: meta/.galaxy_install_info (hunk 1 above)
Comment (StessyMougeot) [reply meta/.galaxy_install_info]: > trash added fro copy/paste

File: tasks/main.yml (hunk 4 above)
Comment (StessyMougeot) [reply tasks/main.yml:48]: Right

File: tasks/main.yml (hunk 5 above)
Comment (StessyMougeot) [reply tasks/main.yml:69]: By default there is no log path, errors are only on stderr

File: tasks/main.yml (hunk 6 above)
Comment (StessyMougeot) [reply tasks/main.yml]: There is a `bad value` error, but I'll add a check before

File: tasks/main.yml (hunk 9 above)
Comment (StessyMougeot) [reply tasks/main.yml]: Yes it's enough, but I've always heard to put both rules

Comment (StessyMougeot) [reply tasks/main.yml]: Asked sec teeam and both aren't needed as we aren't in deny all by default :)

**Review by sterchelen — COMMENTED** [review]
Ok for me as long as you resolve @SwannCroiset 's comments 👌🏼 

To make our work more fun, could you add a gif to the pr's description, showing how you consider this PR :) ?

File: tasks/main.yml (hunk 8 above)
Comment (Dragonnblack) [reply tasks/main.yml]: Right, it isn't present on priv env, check added. 

Need to think how to do it in these envs

File: defaults/main.yml [hunk 10]
```diff
@@ -5,6 +5,7 @@ fluentbit_prerequisites:
   - curl
   - gnupg
 
+fluentbit_groups: "adm"
```
Comment (SwannCroiset) [inline defaults/main.yml]: fluentbit_groups is a list, isn't it ?

File: tasks/main.yml (hunk 4 above)
Comment (SwannCroiset) [reply tasks/main.yml:48]: reloading ?

File: molecule/default/tests/test_default.yml [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+            assert d.exists
+
+
+def test_files(host):
+    present = [
+        "/etc/td-agent-bit/td-agent-bit.conf",
+        "/etc/td-agent-bit/functions.lua",
+        "/etc/systemd/system/td-agent-bit/override.conf"
+    ]
+    if present:
```
Comment (SwannCroiset) [inline molecule/default/tests/test_default.yml]: nit: `if present:` is not necessary

File: defaults/main.yml (hunk 10 above)
Comment (StessyMougeot) [reply defaults/main.yml]: It's a string separated by `,`, better to make a list and a join!

File: defaults/main.yml [hunk 12]
```diff
@@ -14,23 +16,31 @@ fluentbit_svc_name: "td-agent-bit"
...
 
-fluentbit_svc_http:
-  server: "off"
+# Storage
+fluentbit_dbs_path: '/var/run/fluentbit/'
 fluentbit_svc_storage:
-  metrics: "on"
+  Path: '/var/run/fluentbit/fluentbit.buffer'
+  Sync: normal
+  Backlog.mem_limit: '10M'
```
Comment (sterchelen) [inline defaults/main.yml]: Is upper case mandatory? I find upper case variables error prone

File: tasks/main.yml (hunk 8 above)
Comment (sterchelen) [reply tasks/main.yml]: As per swann's comment, is it ok?

File: defaults/main.yml (hunk 12 above)
Comment (StessyMougeot) [reply defaults/main.yml]: Nope, corrected

**Review by RymRabehi — APPROVED** [review]
LGTM

**Review by sterchelen — APPROVED** [review]

**Pull request merged**